{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "naval-sherman",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: monocle3\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    anyDuplicated, append, as.data.frame, basename, cbind, colnames,\n",
      "    dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,\n",
      "    grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,\n",
      "    order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,\n",
      "    rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,\n",
      "    union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "Loading required package: SingleCellExperiment\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘matrixStats’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:Biobase’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:base’:\n",
      "\n",
      "    expand.grid\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "\n",
      "Attaching package: ‘monocle3’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:Biobase’:\n",
      "\n",
      "    exprs, fData, fData<-, pData, pData<-\n",
      "\n",
      "\n",
      "Loading required package: Gviz\n",
      "\n",
      "Loading required package: grid\n",
      "\n",
      "Warning message:\n",
      "\"package 'ggplot2' was built under R version 4.0.3\"\n",
      "Registered S3 method overwritten by 'GGally':\n",
      "  method from   \n",
      "  +.gg   ggplot2\n",
      "\n",
      "Attaching SeuratObject\n",
      "\n",
      "\n",
      "Attaching package: 'Seurat'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:Signac':\n",
      "\n",
      "    FoldChange\n",
      "\n",
      "\n",
      "The following object is masked from 'package:SummarizedExperiment':\n",
      "\n",
      "    Assays\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(cicero)\n",
    "library(monocle3)\n",
    "library(ggplot2)\n",
    "library(patchwork)\n",
    "library(stringr)\n",
    "library(SeuratWrappers)\n",
    "library(Signac)\n",
    "library(Seurat)\n",
    "set.seed(1234)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "chubby-kuwait",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: BSgenome\n",
      "\n",
      "Loading required package: Biostrings\n",
      "\n",
      "Loading required package: XVector\n",
      "\n",
      "Warning message:\n",
      "\"package 'XVector' was built under R version 4.0.3\"\n",
      "\n",
      "Attaching package: 'Biostrings'\n",
      "\n",
      "\n",
      "The following object is masked from 'package:base':\n",
      "\n",
      "    strsplit\n",
      "\n",
      "\n",
      "Loading required package: rtracklayer\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(BSgenome.Mmusculus.UCSC.mm10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "independent-procurement",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "An object of class Seurat \n",
       "184181 features across 3512 samples within 3 assays \n",
       "Active assay: ATAC (157203 features, 157203 variable features)\n",
       " 2 other assays present: ACTIVITY, RNA\n",
       " 2 dimensional reductions calculated: lsi, umap"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "#Load data from previous notebook.\n",
    "brain <- readRDS(\"./adult_mouse_brain/Outputs/brain_coaccs.rds\")\n",
    "brain"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "legislative-hepatitis",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Compute the GC content, region lengths, and dinucleotide base frequencies for regions in the assay and add to the feature metadata.\n",
    "brain <- RegionStats(brain, assay=\"ATAC\", genome = BSgenome.Mmusculus.UCSC.mm10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "ranking-sterling",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Testing 4968 genes and 154435 peaks\n",
      "\n"
     ]
    }
   ],
   "source": [
    "linkpeaks <- LinkPeaks(brain, peak.assay = \"ATAC\", expression.assay = \"RNA\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "id": "animal-budget",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 6282538 ranges and 2 metadata columns:\n",
       "            seqnames             ranges strand |     score     group\n",
       "               <Rle>          <IRanges>  <Rle> | <numeric> <numeric>\n",
       "        [1]     chr1 99773298-100016559      * |  0.402159       854\n",
       "        [2]     chr1 99781706-100016559      * |  0.405137       853\n",
       "        [3]     chr1 99858152-100016559      * |  0.396519       853\n",
       "        [4]     chr1 99872911-100016559      * |  0.226847       854\n",
       "        [5]     chr1 99878040-100016559      * |  0.011585       854\n",
       "        ...      ...                ...    ... .       ...       ...\n",
       "  [6282534]     chrY  90741933-90810851      * | 0.1289795        NA\n",
       "  [6282535]     chrY  90761609-90810851      * | 0.0314409        NA\n",
       "  [6282536]     chrY  90800624-90810851      * | 0.0680179        NA\n",
       "  [6282537]     chrY  90804978-90810851      * | 0.2696632        NA\n",
       "  [6282538]     chrY  90808828-90810851      * | 0.0505001        NA\n",
       "  -------\n",
       "  seqinfo: 21 sequences from an unspecified genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Links(brain)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "id": "happy-graphic",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataFrame with 26755 rows and 5 columns\n",
       "          score        gene                   peak    zscore      pvalue\n",
       "      <numeric> <character>            <character> <numeric>   <numeric>\n",
       "1     0.3308978       Oprk1   chr1-5249384-5249740  12.95154 1.15138e-38\n",
       "2     0.0829117       Oprk1   chr1-5297643-5297838   2.93056 1.69174e-03\n",
       "3     0.0867884       Oprk1   chr1-5382268-5382864   4.00851 3.05510e-05\n",
       "4     0.2777993       Oprk1   chr1-5588241-5588629  12.87920 2.94728e-38\n",
       "5     0.0638828       Oprk1   chr1-5915260-5915954   2.63255 4.23734e-03\n",
       "...         ...         ...                    ...       ...         ...\n",
       "26751  0.428167        Tlr7 chrX-167227198-16722..   3.72202 9.88186e-05\n",
       "26752  0.424684        Tlr7 chrX-167229187-16722..   3.03843 1.18906e-03\n",
       "26753  0.189737        Tlr7 chrX-167231410-16723..   3.19157 7.07503e-04\n",
       "26754  0.405713        Tlr7 chrX-167318905-16731..   5.91863 1.62313e-09\n",
       "26755  0.217494        Tlr7 chrX-167328585-16733..   3.28463 5.10588e-04"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "Links(linkpeaks)@elementMetadata#@listData"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "overhead-robertson",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "GRanges object with 26755 ranges and 5 metadata columns:\n",
       "          seqnames              ranges strand |     score        gene\n",
       "             <Rle>           <IRanges>  <Rle> | <numeric> <character>\n",
       "      [1]     chr1     5249562-5588466      * | 0.3308978       Oprk1\n",
       "      [2]     chr1     5297741-5588466      * | 0.0829117       Oprk1\n",
       "      [3]     chr1     5382566-5588466      * | 0.0867884       Oprk1\n",
       "      [4]     chr1     5588435-5588466      * | 0.2777993       Oprk1\n",
       "      [5]     chr1     5588466-5915607      * | 0.0638828       Oprk1\n",
       "      ...      ...                 ...    ... .       ...         ...\n",
       "  [26751]     chrX 167227346-167330558      * |  0.428167        Tlr7\n",
       "  [26752]     chrX 167229423-167330558      * |  0.424684        Tlr7\n",
       "  [26753]     chrX 167231441-167330558      * |  0.189737        Tlr7\n",
       "  [26754]     chrX 167318993-167330558      * |  0.405713        Tlr7\n",
       "  [26755]     chrX 167329684-167330558      * |  0.217494        Tlr7\n",
       "                            peak    zscore      pvalue\n",
       "                     <character> <numeric>   <numeric>\n",
       "      [1]   chr1-5249384-5249740  12.95154 1.15138e-38\n",
       "      [2]   chr1-5297643-5297838   2.93056 1.69174e-03\n",
       "      [3]   chr1-5382268-5382864   4.00851 3.05510e-05\n",
       "      [4]   chr1-5588241-5588629  12.87920 2.94728e-38\n",
       "      [5]   chr1-5915260-5915954   2.63255 4.23734e-03\n",
       "      ...                    ...       ...         ...\n",
       "  [26751] chrX-167227198-16722..   3.72202 9.88186e-05\n",
       "  [26752] chrX-167229187-16722..   3.03843 1.18906e-03\n",
       "  [26753] chrX-167231410-16723..   3.19157 7.07503e-04\n",
       "  [26754] chrX-167318905-16731..   5.91863 1.62313e-09\n",
       "  [26755] chrX-167328585-16733..   3.28463 5.10588e-04\n",
       "  -------\n",
       "  seqinfo: 20 sequences from an unspecified genome; no seqlengths"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "linkpeaks$ATAC@links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "loving-navigator",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
      "  ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. ..@ values         : Factor w/ 21 levels \"chr1\",\"chr2\",..: 1 10 11 12 13 14 15 16 17 18 ...\n",
      "  .. .. ..@ lengths        : int [1:21] 426572 309382 440854 302976 298440 251138 297776 231574 249374 205382 ...\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
      "  .. .. ..@ start          : int [1:6282538] 99773298 99781706 99858152 99872911 99878040 99880108 99913187 99937111 99985573 100016559 ...\n",
      "  .. .. ..@ width          : int [1:6282538] 243262 234854 158408 143649 138520 136452 103373 79449 30987 30281 ...\n",
      "  .. .. ..@ NAMES          : NULL\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 3\n",
      "  .. .. ..@ lengths        : int 6282538\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
      "  .. .. ..@ seqnames   : chr [1:21] \"chr1\" \"chr2\" \"chr3\" \"chr4\" ...\n",
      "  .. .. ..@ seqlengths : int [1:21] NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. .. ..@ is_circular: logi [1:21] NA NA NA NA NA NA ...\n",
      "  .. .. ..@ genome     : chr [1:21] NA NA NA NA ...\n",
      "  ..@ elementMetadata:Formal class 'DFrame' [package \"S4Vectors\"] with 6 slots\n",
      "  .. .. ..@ rownames       : NULL\n",
      "  .. .. ..@ nrows          : int 6282538\n",
      "  .. .. ..@ listData       :List of 2\n",
      "  .. .. .. ..$ score: num [1:6282538] 0.4022 0.4051 0.3965 0.2268 0.0116 ...\n",
      "  .. .. .. ..$ group: num [1:6282538] 854 853 853 854 854 856 868 856 856 855 ...\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ elementType    : chr \"ANY\"\n",
      "  ..@ metadata       : list()\n"
     ]
    }
   ],
   "source": [
    "str(brain$ATAC@links)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "driven-victim",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Formal class 'Assay' [package \"SeuratObject\"] with 8 slots\n",
      "  ..@ counts       :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. ..@ i       : int [1:26705146] 1 2 5 6 9 10 11 12 13 14 ...\n",
      "  .. .. ..@ p       : int [1:3513] 0 12743 19423 27676 37468 47160 53750 65784 75771 88406 ...\n",
      "  .. .. ..@ Dim     : int [1:2] 21978 3512\n",
      "  .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. ..$ : chr [1:21978] \"Hnf4g\" \"Zfhx4\" \"Pex2\" \"UBC\" ...\n",
      "  .. .. .. ..$ : chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. ..@ x       : num [1:26705146] 2 1 4 1 4 4 4 10 1 13 ...\n",
      "  .. .. ..@ factors : list()\n",
      "  ..@ data         :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. ..@ i       : int [1:26705146] 1 2 5 6 9 10 11 12 13 14 ...\n",
      "  .. .. ..@ p       : int [1:3513] 0 12743 19423 27676 37468 47160 53750 65784 75771 88406 ...\n",
      "  .. .. ..@ Dim     : int [1:2] 21978 3512\n",
      "  .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. ..$ : chr [1:21978] \"Hnf4g\" \"Zfhx4\" \"Pex2\" \"UBC\" ...\n",
      "  .. .. .. ..$ : chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. ..@ x       : num [1:26705146] 0.49 0.275 0.818 0.275 0.818 ...\n",
      "  .. .. ..@ factors : list()\n",
      "  ..@ scale.data   : num[0 , 0 ] \n",
      "  ..@ key          : chr \"activity_\"\n",
      "  ..@ assay.orig   : NULL\n",
      "  ..@ var.features : logi(0) \n",
      "  ..@ meta.features:'data.frame':\t21978 obs. of  0 variables\n",
      "  ..@ misc         : list()\n"
     ]
    }
   ],
   "source": [
    "str(brain$ACTIVITY)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "virtual-camping",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
      "  ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. ..@ values         : Factor w/ 22 levels \"chr3\",\"chrX\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. .. ..@ lengths        : int [1:22] 83099 68561 56688 128222 149106 89406 49376 121281 98969 10906 ...\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
      "  .. .. ..@ start          : int [1:1763965] 3508030 3634150 3638059 3638059 3641223 3641223 3642964 3642964 3644536 3644536 ...\n",
      "  .. .. ..@ width          : int [1:1763965] 303 198 172 172 95 95 107 107 156 156 ...\n",
      "  .. .. ..@ NAMES          : chr [1:1763965] \"ENSMUSE00001236884\" \"ENSMUSE00000676606\" \"ENSMUSE00001345708\" \"ENSMUSE00001345708\" ...\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2 1 2 1 2 1 2 1 2 ...\n",
      "  .. .. ..@ lengths        : int [1:65507] 69 30 1 6 28 37 15 87 16 1 ...\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
      "  .. .. ..@ seqnames   : chr [1:22] \"chr3\" \"chrX\" \"chr16\" \"chr7\" ...\n",
      "  .. .. ..@ seqlengths : int [1:22] 160039680 171031299 98207768 145441459 122082543 149736546 120421639 156508116 124595110 91744698 ...\n",
      "  .. .. ..@ is_circular: logi [1:22] FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      "  .. .. ..@ genome     : chr [1:22] \"mm10\" \"mm10\" \"mm10\" \"mm10\" ...\n",
      "  ..@ elementMetadata:Formal class 'DFrame' [package \"S4Vectors\"] with 6 slots\n",
      "  .. .. ..@ rownames       : NULL\n",
      "  .. .. ..@ nrows          : int 1763965\n",
      "  .. .. ..@ listData       :List of 5\n",
      "  .. .. .. ..$ tx_id       : chr [1:1763965] \"ENSMUST00000108393\" \"ENSMUST00000108394\" \"ENSMUST00000108393\" \"ENSMUST00000108394\" ...\n",
      "  .. .. .. ..$ gene_name   : chr [1:1763965] \"Hnf4g\" \"Hnf4g\" \"Hnf4g\" \"Hnf4g\" ...\n",
      "  .. .. .. ..$ gene_id     : chr [1:1763965] \"ENSMUSG00000017688\" \"ENSMUSG00000017688\" \"ENSMUSG00000017688\" \"ENSMUSG00000017688\" ...\n",
      "  .. .. .. ..$ gene_biotype: chr [1:1763965] \"protein_coding\" \"protein_coding\" \"protein_coding\" \"protein_coding\" ...\n",
      "  .. .. .. ..$ type        : Factor w/ 4 levels \"cds\",\"exon\",\"gap\",..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ elementMetadata: NULL\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ elementType    : chr \"ANY\"\n",
      "  ..@ metadata       : list()\n"
     ]
    }
   ],
   "source": [
    "str(brain$ATAC@annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "exciting-electron",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ChromatinAssay data with 157203 features for 3512 cells\n",
       "Variable features: 157203 \n",
       "Genome: mm10 \n",
       "Annotation present: TRUE \n",
       "Motifs present: FALSE \n",
       "Fragment files: 1 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "brain$ATAC"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "controlling-tutorial",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Formal class 'ChromatinAssay' [package \"Signac\"] with 16 slots\n",
      "  ..@ ranges            :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
      "  .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. .. .. ..@ values         : Factor w/ 21 levels \"chr1\",\"chr2\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. .. .. .. ..@ lengths        : int [1:21] 11406 12381 8105 9894 10228 8982 8871 8060 9029 7953 ...\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
      "  .. .. .. .. ..@ start          : int [1:157203] 3094708 3119414 3204809 3217330 3228123 3292446 3297527 3299618 3309845 3322390 ...\n",
      "  .. .. .. .. ..@ width          : int [1:157203] 845 2369 370 30 341 920 205 402 602 498 ...\n",
      "  .. .. .. .. ..@ NAMES          : NULL\n",
      "  .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 3\n",
      "  .. .. .. .. ..@ lengths        : int 157203\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
      "  .. .. .. .. ..@ seqnames   : chr [1:21] \"chr1\" \"chr2\" \"chr3\" \"chr4\" ...\n",
      "  .. .. .. .. ..@ seqlengths : int [1:21] NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. .. .. .. ..@ is_circular: logi [1:21] NA NA NA NA NA NA ...\n",
      "  .. .. .. .. ..@ genome     : chr [1:21] NA NA NA NA ...\n",
      "  .. .. ..@ elementMetadata:Formal class 'DFrame' [package \"S4Vectors\"] with 6 slots\n",
      "  .. .. .. .. ..@ rownames       : NULL\n",
      "  .. .. .. .. ..@ nrows          : int 157203\n",
      "  .. .. .. .. ..@ listData       : Named list()\n",
      "  .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ motifs            : NULL\n",
      "  ..@ fragments         :List of 1\n",
      "  .. ..$ :Formal class 'Fragment' [package \"Signac\"] with 3 slots\n",
      "  .. .. .. ..@ path : chr \"/home/lorena/Documents/Lorena_Thesis/ATAC/signac/adult_mouse_brain/Inputs/ATAC-seq/atac_v1_adult_brain_fresh_5k\"| __truncated__\n",
      "  .. .. .. ..@ hash : chr [1:2] \"b99ae56bf8a342da814131419fc3b8f8\" \"089ab313205072e3fe1f6e2948895aef\"\n",
      "  .. .. .. ..@ cells: Named chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. .. .. ..- attr(*, \"names\")= chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. .. .. ..- attr(*, \".match.hash\")=Class 'match.hash' <externalptr> \n",
      "  ..@ seqinfo           :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
      "  .. .. ..@ seqnames   : chr [1:66] \"chr1\" \"chr2\" \"chr3\" \"chr4\" ...\n",
      "  .. .. ..@ seqlengths : int [1:66] 195471971 182113224 160039680 156508116 151834684 149736546 145441459 129401213 124595110 130694993 ...\n",
      "  .. .. ..@ is_circular: logi [1:66] FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      "  .. .. ..@ genome     : chr [1:66] \"mm10\" \"mm10\" \"mm10\" \"mm10\" ...\n",
      "  ..@ annotation        :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
      "  .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. .. .. ..@ values         : Factor w/ 22 levels \"chr3\",\"chrX\",..: 1 2 3 4 5 6 7 8 9 10 ...\n",
      "  .. .. .. .. ..@ lengths        : int [1:22] 83099 68561 56688 128222 149106 89406 49376 121281 98969 10906 ...\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
      "  .. .. .. .. ..@ start          : int [1:1763965] 3508030 3634150 3638059 3638059 3641223 3641223 3642964 3642964 3644536 3644536 ...\n",
      "  .. .. .. .. ..@ width          : int [1:1763965] 303 198 172 172 95 95 107 107 156 156 ...\n",
      "  .. .. .. .. ..@ NAMES          : chr [1:1763965] \"ENSMUSE00001236884\" \"ENSMUSE00000676606\" \"ENSMUSE00001345708\" \"ENSMUSE00001345708\" ...\n",
      "  .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 1 2 1 2 1 2 1 2 1 2 ...\n",
      "  .. .. .. .. ..@ lengths        : int [1:65507] 69 30 1 6 28 37 15 87 16 1 ...\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
      "  .. .. .. .. ..@ seqnames   : chr [1:22] \"chr3\" \"chrX\" \"chr16\" \"chr7\" ...\n",
      "  .. .. .. .. ..@ seqlengths : int [1:22] 160039680 171031299 98207768 145441459 122082543 149736546 120421639 156508116 124595110 91744698 ...\n",
      "  .. .. .. .. ..@ is_circular: logi [1:22] FALSE FALSE FALSE FALSE FALSE FALSE ...\n",
      "  .. .. .. .. ..@ genome     : chr [1:22] \"mm10\" \"mm10\" \"mm10\" \"mm10\" ...\n",
      "  .. .. ..@ elementMetadata:Formal class 'DFrame' [package \"S4Vectors\"] with 6 slots\n",
      "  .. .. .. .. ..@ rownames       : NULL\n",
      "  .. .. .. .. ..@ nrows          : int 1763965\n",
      "  .. .. .. .. ..@ listData       :List of 5\n",
      "  .. .. .. .. .. ..$ tx_id       : chr [1:1763965] \"ENSMUST00000108393\" \"ENSMUST00000108394\" \"ENSMUST00000108393\" \"ENSMUST00000108394\" ...\n",
      "  .. .. .. .. .. ..$ gene_name   : chr [1:1763965] \"Hnf4g\" \"Hnf4g\" \"Hnf4g\" \"Hnf4g\" ...\n",
      "  .. .. .. .. .. ..$ gene_id     : chr [1:1763965] \"ENSMUSG00000017688\" \"ENSMUSG00000017688\" \"ENSMUSG00000017688\" \"ENSMUSG00000017688\" ...\n",
      "  .. .. .. .. .. ..$ gene_biotype: chr [1:1763965] \"protein_coding\" \"protein_coding\" \"protein_coding\" \"protein_coding\" ...\n",
      "  .. .. .. .. .. ..$ type        : Factor w/ 4 levels \"cds\",\"exon\",\"gap\",..: 2 2 2 2 2 2 2 2 2 2 ...\n",
      "  .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ bias              : NULL\n",
      "  ..@ positionEnrichment:List of 1\n",
      "  .. ..$ TSS:Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. .. ..@ i       : int [1:6343600] 0 1 2 3 4 6 8 11 13 14 ...\n",
      "  .. .. .. ..@ p       : int [1:2002] 0 2050 4101 6415 8627 10901 13200 15409 17666 19920 ...\n",
      "  .. .. .. ..@ Dim     : int [1:2] 3512 2001\n",
      "  .. .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. .. ..$ : chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. .. .. ..$ : chr [1:2001] \"-999\" \"-998\" \"-997\" \"-996\" ...\n",
      "  .. .. .. ..@ x       : num [1:6343600] 0.855 0.81 1.058 0.936 0.672 ...\n",
      "  .. .. .. ..@ factors : list()\n",
      "  ..@ links             :Formal class 'GRanges' [package \"GenomicRanges\"] with 7 slots\n",
      "  .. .. ..@ seqnames       :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. .. .. ..@ values         : Factor w/ 21 levels \"chr1\",\"chr2\",..: 1 10 11 12 13 14 15 16 17 18 ...\n",
      "  .. .. .. .. ..@ lengths        : int [1:21] 426572 309382 440854 302976 298440 251138 297776 231574 249374 205382 ...\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ ranges         :Formal class 'IRanges' [package \"IRanges\"] with 6 slots\n",
      "  .. .. .. .. ..@ start          : int [1:6282538] 99773298 99781706 99858152 99872911 99878040 99880108 99913187 99937111 99985573 100016559 ...\n",
      "  .. .. .. .. ..@ width          : int [1:6282538] 243262 234854 158408 143649 138520 136452 103373 79449 30987 30281 ...\n",
      "  .. .. .. .. ..@ NAMES          : NULL\n",
      "  .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ strand         :Formal class 'Rle' [package \"S4Vectors\"] with 4 slots\n",
      "  .. .. .. .. ..@ values         : Factor w/ 3 levels \"+\",\"-\",\"*\": 3\n",
      "  .. .. .. .. ..@ lengths        : int 6282538\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ seqinfo        :Formal class 'Seqinfo' [package \"GenomeInfoDb\"] with 4 slots\n",
      "  .. .. .. .. ..@ seqnames   : chr [1:21] \"chr1\" \"chr2\" \"chr3\" \"chr4\" ...\n",
      "  .. .. .. .. ..@ seqlengths : int [1:21] NA NA NA NA NA NA NA NA NA NA ...\n",
      "  .. .. .. .. ..@ is_circular: logi [1:21] NA NA NA NA NA NA ...\n",
      "  .. .. .. .. ..@ genome     : chr [1:21] NA NA NA NA ...\n",
      "  .. .. ..@ elementMetadata:Formal class 'DFrame' [package \"S4Vectors\"] with 6 slots\n",
      "  .. .. .. .. ..@ rownames       : NULL\n",
      "  .. .. .. .. ..@ nrows          : int 6282538\n",
      "  .. .. .. .. ..@ listData       :List of 2\n",
      "  .. .. .. .. .. ..$ score: num [1:6282538] 0.4022 0.4051 0.3965 0.2268 0.0116 ...\n",
      "  .. .. .. .. .. ..$ group: num [1:6282538] 854 853 853 854 854 856 868 856 856 855 ...\n",
      "  .. .. .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. .. .. ..@ elementMetadata: NULL\n",
      "  .. .. .. .. ..@ metadata       : list()\n",
      "  .. .. ..@ elementType    : chr \"ANY\"\n",
      "  .. .. ..@ metadata       : list()\n",
      "  ..@ counts            :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. ..@ i       : int [1:45482860] 1 4 20 28 30 32 33 40 48 49 ...\n",
      "  .. .. ..@ p       : int [1:3513] 0 32050 42572 57343 71020 91393 102892 130830 151099 182121 ...\n",
      "  .. .. ..@ Dim     : int [1:2] 157203 3512\n",
      "  .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. ..$ : chr [1:157203] \"chr1-3094708-3095552\" \"chr1-3119414-3121782\" \"chr1-3204809-3205178\" \"chr1-3217330-3217359\" ...\n",
      "  .. .. .. ..$ : chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. .. .. ..- attr(*, \".match.hash\")=Class 'match.hash' <externalptr> \n",
      "  .. .. ..@ x       : num [1:45482860] 2 2 2 1 8 2 1 2 4 4 ...\n",
      "  .. .. ..@ factors : list()\n",
      "  ..@ data              :Formal class 'dgCMatrix' [package \"Matrix\"] with 6 slots\n",
      "  .. .. ..@ i       : int [1:45482860] 1 4 20 28 30 32 33 40 48 49 ...\n",
      "  .. .. ..@ p       : int [1:3513] 0 32050 42572 57343 71020 91393 102892 130830 151099 182121 ...\n",
      "  .. .. ..@ Dim     : int [1:2] 157203 3512\n",
      "  .. .. ..@ Dimnames:List of 2\n",
      "  .. .. .. ..$ : chr [1:157203] \"chr1-3094708-3095552\" \"chr1-3119414-3121782\" \"chr1-3204809-3205178\" \"chr1-3217330-3217359\" ...\n",
      "  .. .. .. ..$ : chr [1:3512] \"AAACGAAAGTAATCAG-1\" \"AAACGAACACGCTGTG-1\" \"AAACGAATCCTGGGAC-1\" \"AAACGAATCGGGAAAC-1\" ...\n",
      "  .. .. .. .. ..- attr(*, \".match.hash\")=Class 'match.hash' <externalptr> \n",
      "  .. .. ..@ x       : num [1:45482860] 0.351 1.93 0.666 0.894 0.403 ...\n",
      "  .. .. ..@ factors : list()\n",
      "  ..@ scale.data        : num[0 , 0 ] \n",
      "  ..@ key               : chr \"atac_\"\n",
      "  ..@ assay.orig        : NULL\n",
      "  ..@ var.features      : chr [1:157203] \"chr17-39842772-39849053\" \"chr1-172305860-172331966\" \"chr9-57071113-57077709\" \"chr13-83712987-83734344\" ...\n",
      "  ..@ meta.features     :'data.frame':\t157203 obs. of  2 variables:\n",
      "  .. ..$ count     : num [1:157203] 562 1863 153 11 133 ...\n",
      "  .. ..$ percentile: num [1:157203] 0.7019 0.9039 0.2808 0.0189 0.2306 ...\n",
      "  ..@ misc              : list()\n"
     ]
    }
   ],
   "source": [
    "str(brain$ATAC)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "royal-delay",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "\n",
       "<table width=\"100%\" summary=\"page for GeneActivity {Signac}\"><tr><td>GeneActivity {Signac}</td><td style=\"text-align: right;\">R Documentation</td></tr></table>\n",
       "\n",
       "<h2>Create gene activity matrix</h2>\n",
       "\n",
       "<h3>Description</h3>\n",
       "\n",
       "<p>Compute counts per cell in gene body and promoter region.\n",
       "</p>\n",
       "\n",
       "\n",
       "<h3>Usage</h3>\n",
       "\n",
       "<pre>\n",
       "GeneActivity(\n",
       "  object,\n",
       "  assay = NULL,\n",
       "  features = NULL,\n",
       "  extend.upstream = 2000,\n",
       "  extend.downstream = 0,\n",
       "  verbose = TRUE,\n",
       "  ...\n",
       ")\n",
       "</pre>\n",
       "\n",
       "\n",
       "<h3>Arguments</h3>\n",
       "\n",
       "<table summary=\"R argblock\">\n",
       "<tr valign=\"top\"><td><code>object</code></td>\n",
       "<td>\n",
       "<p>A Seurat object</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>assay</code></td>\n",
       "<td>\n",
       "<p>Name of assay to use. If NULL, use the default assay</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>features</code></td>\n",
       "<td>\n",
       "<p>Genes to include. If NULL, use all protein-coding genes in\n",
       "the annotations stored in the object</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>extend.upstream</code></td>\n",
       "<td>\n",
       "<p>Number of bases to extend upstream of the TSS</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>extend.downstream</code></td>\n",
       "<td>\n",
       "<p>Number of bases to extend downstream of the TTS</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>verbose</code></td>\n",
       "<td>\n",
       "<p>Display messages</p>\n",
       "</td></tr>\n",
       "<tr valign=\"top\"><td><code>...</code></td>\n",
       "<td>\n",
       "<p>Additional options passed to <code>FeatureMatrix</code></p>\n",
       "</td></tr>\n",
       "</table>\n",
       "\n",
       "\n",
       "<h3>Examples</h3>\n",
       "\n",
       "<pre>\n",
       "fpath &lt;- system.file(\"extdata\", \"fragments.tsv.gz\", package=\"Signac\")\n",
       "fragments &lt;- CreateFragmentObject(\n",
       "  path = fpath,\n",
       "  cells = colnames(atac_small),\n",
       "  validate.fragments = FALSE\n",
       ")\n",
       "Fragments(atac_small) &lt;- fragments\n",
       "GeneActivity(atac_small)\n",
       "</pre>\n",
       "\n",
       "<hr /><div style=\"text-align: center;\">[Package <em>Signac</em> version 1.1.1 ]</div>"
      ],
      "text/latex": [
       "\\inputencoding{utf8}\n",
       "\\HeaderA{GeneActivity}{Create gene activity matrix}{GeneActivity}\n",
       "\\keyword{utilities}{GeneActivity}\n",
       "%\n",
       "\\begin{Description}\\relax\n",
       "Compute counts per cell in gene body and promoter region.\n",
       "\\end{Description}\n",
       "%\n",
       "\\begin{Usage}\n",
       "\\begin{verbatim}\n",
       "GeneActivity(\n",
       "  object,\n",
       "  assay = NULL,\n",
       "  features = NULL,\n",
       "  extend.upstream = 2000,\n",
       "  extend.downstream = 0,\n",
       "  verbose = TRUE,\n",
       "  ...\n",
       ")\n",
       "\\end{verbatim}\n",
       "\\end{Usage}\n",
       "%\n",
       "\\begin{Arguments}\n",
       "\\begin{ldescription}\n",
       "\\item[\\code{object}] A Seurat object\n",
       "\n",
       "\\item[\\code{assay}] Name of assay to use. If NULL, use the default assay\n",
       "\n",
       "\\item[\\code{features}] Genes to include. If NULL, use all protein-coding genes in\n",
       "the annotations stored in the object\n",
       "\n",
       "\\item[\\code{extend.upstream}] Number of bases to extend upstream of the TSS\n",
       "\n",
       "\\item[\\code{extend.downstream}] Number of bases to extend downstream of the TTS\n",
       "\n",
       "\\item[\\code{verbose}] Display messages\n",
       "\n",
       "\\item[\\code{...}] Additional options passed to \\code{\\LinkA{FeatureMatrix}{FeatureMatrix}}\n",
       "\\end{ldescription}\n",
       "\\end{Arguments}\n",
       "%\n",
       "\\begin{Examples}\n",
       "\\begin{ExampleCode}\n",
       "fpath <- system.file(\"extdata\", \"fragments.tsv.gz\", package=\"Signac\")\n",
       "fragments <- CreateFragmentObject(\n",
       "  path = fpath,\n",
       "  cells = colnames(atac_small),\n",
       "  validate.fragments = FALSE\n",
       ")\n",
       "Fragments(atac_small) <- fragments\n",
       "GeneActivity(atac_small)\n",
       "\\end{ExampleCode}\n",
       "\\end{Examples}"
      ],
      "text/plain": [
       "GeneActivity              package:Signac               R Documentation\n",
       "\n",
       "_\bC_\br_\be_\ba_\bt_\be _\bg_\be_\bn_\be _\ba_\bc_\bt_\bi_\bv_\bi_\bt_\by _\bm_\ba_\bt_\br_\bi_\bx\n",
       "\n",
       "_\bD_\be_\bs_\bc_\br_\bi_\bp_\bt_\bi_\bo_\bn:\n",
       "\n",
       "     Compute counts per cell in gene body and promoter region.\n",
       "\n",
       "_\bU_\bs_\ba_\bg_\be:\n",
       "\n",
       "     GeneActivity(\n",
       "       object,\n",
       "       assay = NULL,\n",
       "       features = NULL,\n",
       "       extend.upstream = 2000,\n",
       "       extend.downstream = 0,\n",
       "       verbose = TRUE,\n",
       "       ...\n",
       "     )\n",
       "     \n",
       "_\bA_\br_\bg_\bu_\bm_\be_\bn_\bt_\bs:\n",
       "\n",
       "  object: A Seurat object\n",
       "\n",
       "   assay: Name of assay to use. If NULL, use the default assay\n",
       "\n",
       "features: Genes to include. If NULL, use all protein-coding genes in\n",
       "          the annotations stored in the object\n",
       "\n",
       "extend.upstream: Number of bases to extend upstream of the TSS\n",
       "\n",
       "extend.downstream: Number of bases to extend downstream of the TTS\n",
       "\n",
       " verbose: Display messages\n",
       "\n",
       "     ...: Additional options passed to 'FeatureMatrix'\n",
       "\n",
       "_\bE_\bx_\ba_\bm_\bp_\bl_\be_\bs:\n",
       "\n",
       "     fpath <- system.file(\"extdata\", \"fragments.tsv.gz\", package=\"Signac\")\n",
       "     fragments <- CreateFragmentObject(\n",
       "       path = fpath,\n",
       "       cells = colnames(atac_small),\n",
       "       validate.fragments = FALSE\n",
       "     )\n",
       "     Fragments(atac_small) <- fragments\n",
       "     GeneActivity(atac_small)\n",
       "     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "?GeneActivity"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "r_signac_clone",
   "language": "R",
   "name": "r_signac_clone"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.0.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
